src/main/java/com/icthh/xm/tmf/ms/communication/config/KafkaChannelFactory.java

```diff
@@ -151,4 +152,27 @@ private CommunicationMessage mapToCommunicationMessage(String eventBody) {
         return objectMapper.readValue(eventBody, CommunicationMessage.class);
     }
 
+    /**
+     * Since Kafka headers are not accessible from the business rules,
+     * move Kafka received timestamp to the communication message characteristics
+     */
+    private void addReceivedByChannelCharacteristic(CommunicationMessage communicationMessage, Message<?> kafkaMessage){
+        // cannot be null, but anyway
+        if (kafkaMessage.getHeaders() != null){
```

Comment:
please consider if it worth to use functional approach with Optional like this:
```
Optional.ofNullable(kafkaMessage)
   .map(Message::getHeaders)
   .map(headers -> headers.get(KafkaHeaders.RECEIVED_TIMESTAMP))
   .map(Long::cast)
   .ifPresent(time -> 
                    communicationMessage.addCharacteristicItem(
                    new CommunicationRequestCharacteristic()
                        // Rename it to unlink name from source channel
                        .name(MESSAGE_RECEIVED_BY_CHANNEL_TIMESTAMP)
                        .value(String.valueOf(kafkaReceivedTimestamp))
                );
)
```

Reply:
Done